Update Web Resources

tcyrus opened the pull request:
- Replaced jQuery 3.5.1 with jQuery 3.7.1 Slim
- Removed Unused Socket.IO 2.3.0 Library
- Replaced Socket.IO 4.4.1 with Socket.IO 4.8.1
- Replaced Some jQuery Calls with Vanilla JS

mig5 reviewed (changes requested):
Sorry for the delay in reviewing. Due to other changes made to `receive.js` are you able to fix the conflict? Apologies for the delay being partly responsible for that.

tcyrus committed:
Update Resources

- Replaced jQuery 3.5.1 with jQuery 3.7.1 Slim
- Removed Unused Socket.IO 2.3.0 Library
- Replaced Socket.IO 4.4.1 with Socket.IO 4.8.1
- Replaced Some jQuery Calls with Vanilla JS

tcyrus commented:
@mig5 sorry for the delay. I completely forgot about this PR. Just rebased the PR branch and resolved the merge conflict.

mig5 reviewed (changes requested):
Thanks @tcyrus !

The regular 'Share' mode looks fine. Unfortunately with Receive mode, a javascript error is thrown when you try and send a message or upload a file:

```
Uncaught TypeError: can't access property "textContent", uploadDivEl.querySelector(...) is null
    <anonymous> http://127.0.0.1:17629/static_xdlleaamcaitbskdoo7hlfal2m/js/receive.js:157
    EventListener.handleEvent* http://127.0.0.1:17629/static_xdlleaamcaitbskdoo7hlfal2m/js/receive.js:18
    jQuery 11
    <anonymous> http://127.0.0.1:17629/static_xdlleaamcaitbskdoo7hlfal2m/js/receive.js:1
receive.js:157:17
```

The same error occurs in Chat mode when you attempt to send a message:

```
Uncaught (in promise) TypeError: can't access property "textContent", messageEl.getElementsByClassName(...)[0] is undefined
    addChatMessage http://127.0.0.1:17634/static_2tjs43kvvr22ktuacwncl6ufyy/js/chat.js:214
    addMessageToPanel http://127.0.0.1:17634/static_2tjs43kvvr22ktuacwncl6ufyy/js/chat.js:68
    <anonymous> http://127.0.0.1:17634/static_2tjs43kvvr22ktuacwncl6ufyy/js/chat.js:24
    emit http://127.0.0.1:17634/static_2tjs43kvvr22ktuacwncl6ufyy/js/socket.io.min.js:6
    emitEvent http://127.0.0.1:17634/static_2tjs43kvvr22ktuacwncl6ufyy/js/socket.io.min.js:6
    onevent http://127.0.0.1:17634/static_2tjs43kvvr22ktuacwncl6ufyy/js/socket.io.min.js:6
    onpacket http://127.0.0.1:17634/static_2tjs43kvvr22ktuacwncl6ufyy/js/socket.io.min.js:6
    emit http://127.0.0.1:17634/static_2tjs43kvvr22ktuacwncl6ufyy/js/socket.io.min.js:6
    ondecoded http://127.0.0.1:17634/static_2tjs43kvvr22ktuacwncl6ufyy/js/socket.io.min.js:6
    promise callback*R< http://127.0.0.1:17634/static_2tjs43kvvr22ktuacwncl6ufyy/js/socket.io.min.js:6
    ondecoded http://127.0.0.1:17634/static_2tjs43kvvr22ktuacwncl6ufyy/js/socket.io.min.js:6
    emit http://127.0.0.1:17634/static_2tjs43kvvr22ktuacwncl6ufyy/js/socket.io.min.js:6
    add http://127.0.0.1:17634/static_2tjs43kvvr22ktuacwncl6ufyy/js/socket.io.min.js:6
    ondata http://127.0.0.1:17634/static_2tjs43kvvr22ktuacwncl6ufyy/js/socket.io.min.js:6
    emit http://127.0.0.1:17634/static_2tjs43kvvr22ktuacwncl6ufyy/js/socket.io.min.js:6
    H http://127.0.0.1:17634/static_2tjs43kvvr22ktuacwncl6ufyy/js/socket.io.min.js:6
    emit http://127.0.0.1:17634/static_2tjs43kvvr22ktuacwncl6ufyy/js/socket.io.min.js:6
    onPacket http://127.0.0.1:17634/static_2tjs43kvvr22ktuacwncl6ufyy/js/socket.io.min.js:6
    onData http://127.0.0.1:17634/static_2tjs43kvvr22ktuacwncl6ufyy/js/socket.io.min.js:6
    onmessage http://127.0.0.1:17634/static_2tjs43kvvr22ktuacwncl6ufyy/js/socket.io.min.js:6
chat.js:214:13
```

Let me know if you think you have time to figure this out.

tcyrus committed:
Fix js issue

outerHTML didn't work since the element didn't have a parent element at
the time

tcyrus commented:
It would be a good idea to add a `.browserslistrc` file so that the browser target is clearly defined to ensure JavaScript compatibility.

mig5 committed:
Upgrade to Socket.IO v4.8.3

mig5 committed:
Fix inline CSS breaking Receive mode CSP

mig5 reviewed (approved):
Thanks @tcyrus for sticking with us. I bumped SocketIO to the latest version, and fixed an unrelated CSP issue in Receive Mode (that was due to other recent changes there, not yours, which slipped past me in a past review).

Seems to all be working well!

mig5 commented:
> It would be a good idea to add a `.browserslistrc` file so that the browser target is clearly defined to ensure JavaScript compatibility.

We'll look at this in due course, but for now, I'll merge to avoid blocking release. Thanks again
